src/main/java/org/prebid/server/handler/SetuidHandler.java

```diff
@@ -103,13 +105,22 @@ public SetuidHandler(long defaultTimeout,
         this.analyticsDelegator = Objects.requireNonNull(analyticsDelegator);
         this.metrics = Objects.requireNonNull(metrics);
         this.timeoutFactory = Objects.requireNonNull(timeoutFactory);
+        this.cookieNameToSyncType = collectMap(bidderCatalog);
+    }
+
+    private static Map<String, UsersyncMethodType> collectMap(BidderCatalog bidderCatalog) {
 
-        cookieNameToSyncType = bidderCatalog.names().stream()
+        final Supplier<Stream<Usersyncer>> usersyncers = () -> bidderCatalog.names()
+                .stream()
                 .filter(bidderCatalog::isActive)
                 .map(bidderCatalog::usersyncerByName)
                 .filter(Optional::isPresent)
                 .map(Optional::get)
-                .distinct() // built-in aliases looks like bidders with the same usersyncers
+                .distinct();
+
+        validateUsersyncers(usersyncers.get());
+
+        return usersyncers.get()
```

Comment:
What about this approach?
```
bidderCatalog.names().stream()
                .filter(bidderCatalog::isActive)
                .map(bidderCatalog::usersyncerByName)
                .filter(Optional::isPresent)
                .map(Optional::get)
                .distinct() // built-in aliases looks like bidders with the same usersyncers
                .filter(usersyncValidator())
                .collect(Collectors.toMap(Usersyncer::getCookieFamilyName, SetuidHandler::preferredUserSyncType));
```
```
    private static Predicate<Usersyncer> usersyncValidator() {
        final Set<String> cookieFamilyNames = new HashSet<>();
        return usersyncer -> {
            if (!cookieFamilyNames.add(usersyncer.getCookieFamilyName())) {
                throw new IllegalArgumentException(...);
            }
            return true;
        };
    }
```

Reply:
Indeed, looks much better. But we should have trade off. Or we choose more simple solution, but without full list of duplicates, or vice-versa.

I preffer this one. @Net-burst your call.

Reply:
I don't like the approach to throw an exception from the predicate, it's not intuitive.  

Looks like the suggestion above will fail on the first duplicate, but if we need to printout all the duplicates in the config, the presented approach is better.